conformance/lang/values-types-and-variables/map_types.balt

```diff
@@ -0,0 +1,211 @@
+                   // @output 2
+                   // @output 0 
+}
+
+Test-case: error
+Description: Test the type of the values in the iteration sequence of a value belonging to map<T> being T 
+             via incompatible assignments.
+Labels: map-type-descriptor, mapping-type-descriptor, string-type-descriptor, int-type-descriptor, 
+        mapping-constructor-expr, int-literal, foreach-stmt
+
+public function main() {
+    map<int> a = {a: 1, b: 2};
+
+    foreach string s in a { // @error value in the iteration sequence should be int
+
+    }
+}
+
+Test-case: error
+Description: Test the iteration completion value of a map being nil via incompatible assignments.
+Labels: map-type-descriptor, mapping-type-descriptor, record-type-descriptor, int-type-descriptor, 
+        error-type-descriptor, mapping-constructor-expr, lang.map
+
+public function main() {
+    map<int> a = {};    
+    record {| int value; |}|error next = a.iterator().next(); // @error completion type for map iteration is nil
+}
+
+Test-case: output
+Description: Test mapping type-descriptor being covariant in the types of their members.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, byte-type-descriptor, 
+        anydata-type-descriptor, mapping-constructor-expr, int-literal
+
+public function main() {
+    map<byte> a = {a: 1, b: 2};
+    testMap(a);
+
+    map<int> b = a;
+    testMap(b);
+}
+
+function testMap(map<anydata> m) {
+    io:println(m); // @output {"a":1,"b":2}
+                   // @output {"a":1,"b":2} 
+}
+
+Test-case: output
+Description: Test mapping type-descriptor being covariant in the types of their members, at runtime.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, byte-type-descriptor, 
+        anydata-type-descriptor, mapping-constructor-expr, int-literal, string-literal, is-expr
+
+public function main() {
+    map<byte> a = {a: 1, b: 2};
+    testMap(a);
+
+    map<int> b = a;
+    testMap(b);
+
+    map<int> c = {x: 1, y: 2, z: 3};
+    testMap(c);
+
+    map<anydata> d = {a: 1, b: "hello world"};
+    testMap(d);
+
+    testMap({a: 1, b: 2});
+}
+
+function testMap(map<anydata> m) {
+    io:println(m is map<int>); // @output true
+                               // @output true
+                               // @output true
+                               // @output false
+                               // @output false
+}
+
+Test-case: error
+Description: Test mapping type-descriptor being covariant in the types of their members, via incompatible assignments.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, byte-type-descriptor, 
+        string-type-descriptor, mapping-constructor-expr
+
+public function main() {
+    map<int> a = {};
+    testMap(a); // @error map of int is not a subtype of map of byte
+
+    map<string> b = a; // @error map of int is not a subtype of map of string
+}
+
+function testMap(map<byte> m) {
```

Comment:
I would `int` and `int?` rather than `int` and `byte`. Subtypes of basic types are more complex than unions of basic types.